src/vma/dev/net_device_val.cpp

```diff
@@ -308,18 +342,100 @@ net_device_val::~net_device_val()
 	m_ip.clear();
 }
 
-void net_device_val::set_ip_array(struct ifaddrs* ifa)
+void net_device_val::set_ip_array()
 {
-	ip_data_t* p_val = NULL;
+	int rc = 0;
+	int fd = -1;
+	struct {
+		struct nlmsghdr hdr;
+		struct ifaddrmsg addrmsg;
+	} nl_req;
+	struct nlmsghdr *nl_msg;
+	int nl_msglen = 0;
+	char nl_res[8096];
+	static int _seq = 0;
+
+	/* Set up the netlink socket */
+	fd = orig_os_api.socket(AF_NETLINK, SOCK_RAW, NETLINK_ROUTE);
+	if (fd < 0) {
+		nd_logerr("netlink socket() creation");
+		return;
+	}
+
+	/* Prepare RTM_GETADDR request */
+	memset(&nl_req, 0, sizeof(nl_req));
+	nl_req.hdr.nlmsg_len = NLMSG_LENGTH(sizeof(struct ifaddrmsg));
+	nl_req.hdr.nlmsg_flags = NLM_F_REQUEST | NLM_F_DUMP;
+	nl_req.hdr.nlmsg_type = RTM_GETADDR;
+	nl_req.hdr.nlmsg_seq = _seq++;
+	nl_req.hdr.nlmsg_pid = getpid();
+	nl_req.addrmsg.ifa_family = AF_INET;
+	nl_req.addrmsg.ifa_index = m_if_idx;
+
```

Comment:
Why are we query netlink twice ? i dont understand why we dont using netlink_socket_mgr as the parent of those class...